### PR TITLE
Solved deprecated use of 'Variant' members + Updated pubspec.lock for Flutter 2.10 release

### LINKED ISSUE #1190: `flutter analyze --no-pub` command failing with Flutter 2.10.0
**Describe the bug**
The latest version of Flutter (2.10.0 ) has deprecated the `primaryVariant` and `secondaryVariant` arguments from the [ColorScheme constructor](https://api.flutter.dev/flutter/material/ColorScheme/ColorScheme.html). This is causing our GitHub CI/CD pipeline to fail when running the `flutter analyze --no-pub`

The solution is to update the codebase to:

1. Use `primary` or `primaryContainer` instead of `primaryVariant`
2. Use `secondary` or `secondaryContainer` instead of `secondaryVariant`

The fix looks relatively simple. The issue occurs in 23 locations in 13 files.

**To Reproduce**

1. Run `flutter analyze --no-pub`
2. Observe errors below

**Expected behavior**
The `flutter analyze --no-pub` command must run without 

**Actual behavior**
```
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/constants/custom_theme.dart:35:7 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/constants/custom_theme.dart:38:7 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/constants/custom_theme.dart:55:7 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/constants/custom_theme.dart:58:7 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/view_model/main_screen_view_model.dart:104:50 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/after_auth_screens/events/explore_event_dialogue.dart:50:62 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/after_auth_screens/events/explore_event_dialogue.dart:89:62 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/after_auth_screens/join_organisation_after_auth.dart:75:28 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/after_auth_screens/profile/profile_page.dart:215:66 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/pre_auth_screens/set_url.dart:141:57 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/pre_auth_screens/set_url.dart:155:64 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/pre_auth_screens/waiting_to_join_private_org.dart:90:60 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/views/pre_auth_screens/waiting_to_join_private_org.dart:111:53 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/widgets/event_date_time_tile.dart:20:44 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/widgets/member_name_tile.dart:24:48 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/widgets/pinned_carousel_widget.dart:24:48 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/widgets/pinned_carousel_widget.dart:37:50 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • lib/widgets/post_widget.dart:44:48 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • test/widget_tests/pre_auth_screens/set_url_page_test.dart:285:44 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • test/widget_tests/pre_auth_screens/set_url_page_test.dart:319:44 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • test/widget_tests/pre_auth_screens/set_url_page_test.dart:522:43 • deprecated_member_use
   info • 'secondaryVariant' is deprecated and shouldn't be used. Use secondary or secondaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • test/widget_tests/pre_auth_screens/set_url_page_test.dart:556:43 • deprecated_member_use
   info • 'primaryVariant' is deprecated and shouldn't be used. Use primary or primaryContainer instead. This feature was deprecated after v2.6.0-0.0.pre. • test/widget_tests/widgets/member_name_tile_test.dart:47:52 • deprecated_member_use
```

**Screenshots**
N/A

**Additional details**
N/A

### PR DESCRIPTION
**Pull request by ItsAdityaKSingh**
**What kind of change does this PR introduce?**

Discontinued use of deprecated members and updated pubspec.lock

**Issue Number:**

Fixes #1190 

**Did you add tests for your changes?**

No tests needed

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -51,6 +51,9 @@ dependencies:
   video_player: ^2.2.7
   visibility_detector: ^0.2.2
 
+dependency_overrides:
```
Comment (palisadoes) [inline pubspec.yaml:54]: Why is this necessary? The uses standard `ColorScheme` class.

**Review by palisadoes — CHANGES_REQUESTED** [review]
Review your [pubspec.yaml](url) file. The new dependency seems unnecessary.

File: pubspec.yaml (hunk 1 above)
Comment (ItsAdityaKSingh) [reply pubspec.yaml:54]: It's been added to prevent platform from having errors such as `io.Platform.packageRoot` error. Without it, debugging won't run.

Comment (palisadoes) [reply pubspec.yaml:54]: Please create another issue to resolve the `io.Platform.packageRoot` error

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
Please create an issue to fix the `io.Platform.packageRoot` error

**Comment by ItsAdityaKSingh** [comment]
> Please create an issue to fix the `io.Platform.packageRoot` error

The error has been resolved in this particular commit itself. Still do I need to create an issue @palisadoes ?

**Comment by palisadoes** [comment]
Don't worry. I did some research on this. Thanks for fixing it.

**Comment by ItsAdityaKSingh** [comment]
Glad to be of help @palisadoes :)
